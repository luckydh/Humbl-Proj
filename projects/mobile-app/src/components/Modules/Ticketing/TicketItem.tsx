import moment from "moment";
import React, { FC, useState } from "react";
import { useTranslation } from "react-i18next";

type RefReturn =
  | string
  | ((instance: HTMLSelectElement | null) => void)
  | React.RefObject<HTMLSelectElement>
  | null;

export type TicketItemProps = {
  name: string;
  register: RefReturn;
  title: string;
  description?: string;
  price: string;
  min: number;
  max: number;
  saleEndDate?: Date;
  ticketId?: string;
  badge?: {
    label: string;
    type: string;
  };
};

interface OptionProps {
  name: string;
  register: RefReturn;
  max: number;
  min: number;
}
const Options: FC<OptionProps> = (props) => {
  const optionList = [];
  if (props.min > 0) {
    optionList.push(<option key={0}>0</option>);
  }
  for (let index = props.min; index <= props.max; index+=1) {
    optionList.push(<option key={index}>{index}</option>);
  }
  return (
    <select
      name={props.name}
      ref={props.register}
      style={{ height: 47, fontSize: 20, minWidth: 84 }}
      className="py-2 px-4 w-full rounded-lg font-semibold bg-blue border-white border-1 outline-none text-white placeholder-white-faded text-xl block focus:ring-indigo-500 focus:border-white focus:border-transparent focus:outline-none focus:ring-0 focus:placeholder-transparent"
    >
      {optionList}
    </select>
  );
};

export const TicketItem: FC<TicketItemProps> = ({
  name,
  register,
  title,
  description,
  price,
  min,
  max,
  badge,
  saleEndDate,
}) => {
  const { t } = useTranslation();

  const [showMore, setShowMore] = useState(false);
  const toggleDescription = () => {
    setShowMore(!showMore);
  };

  return (
    <div className="border-b border-gray-200 mb-6 pb-6 text-white">
      <div>
        <div className="flex w-full justify-between">
          <div className="mr-4" style={{ maxWidth: "13rem" }}>
            <h3 className="font-bold text-xl"> {title}</h3>
            <h4 className="text-lg my-1">{price}</h4>
            {saleEndDate && (
              <p className="text-base">
                {t("ticketing.component.ticket_item.sales_end_on")}{" "}
                {moment(saleEndDate).format("dddd, MMM D, YYYY")}
              </p>
            )}
            <div>
              {!!badge && badge.type === "ticketWarning" && (
                <span className="text-red text-sm">{badge.label}</span>
              )}
            </div>
          </div>
          <div>
            {!!badge && badge.type === "warning" && (
              <span className="inline-flex items-center px-4 py-1 rounded-full text-base font-medium bg-red-light text-red">
                {badge.label}
              </span>
            )}
            {(!badge || (!!badge && badge.type !== "warning")) && (
              <Options
                min={min}
                max={max}
                name={name}
                register={register}
              />
            )}
          </div>
        </div>
        <div>
          {description && (
            <>
              {showMore && <p className="my-2">{description}</p>}
              <button
                className="text-base text-blue-dark font-bold underline mt-2"
                onClick={toggleDescription}
              >
                {!showMore
                  ? t("ticketing.component.ticket_item.see-more")
                  : t("ticketing.component.ticket_item.see_less")}
              </button>
            </>
          )}
        </div>
      </div>
    </div>
  );
};

export default TicketItem;
TicketItem.displayName = "TicketItem";
