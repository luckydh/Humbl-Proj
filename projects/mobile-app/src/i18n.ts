import i18n from "i18next";
import { initReactI18next, useTranslation } from "react-i18next";
import { register } from "timeago.js";
import { es, pt_BR } from "timeago.js/lib/lang";
import HttpApi from "i18next-http-backend";
import LanguageDetector from "i18next-browser-languagedetector";
//import resources from "./locales/index";
// resources: resources,

export type TFnType = ReturnType<typeof useTranslation>["t"];

register("es", es);
register("pt", pt_BR);

i18n
  .use(HttpApi)
  .use(LanguageDetector)
  .use(initReactI18next) // passes i18n down to react-i18next
  .init({
    backend: {
      loadPath: "https://api.locize.app/f4e7a8d0-8642-44bb-ad02-ce1d5554b83c/latest/{{lng}}/{{ns}}",
      addPath: "https://api.locize.app/f4e7a8d0-8642-44bb-ad02-ce1d5554b83c/latest/add/{{lng}}/{{ns}}",
      allowMultiLoading: false,
    },
    //resources,
    //If we don't have a transalation and only have empty string, display fallback
    returnEmptyString: false,
    fallbackLng: "en",
    debug: false,
    ns: "translation",
    keySeparator: false, // we do not use keys in form messages.welcome
    interpolation: {
      escapeValue: false, // react already safes from xss
    },
  });

export default i18n;
