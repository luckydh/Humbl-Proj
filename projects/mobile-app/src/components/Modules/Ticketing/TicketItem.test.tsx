import i18n from "i18n";
import React from "react";
import renderer from "react-test-renderer";
import TicketItem from "./TicketItem";

//Disable suspense on menu drawer

if (i18n.options) {
  i18n.options.react = {
    useSuspense: false,
  };
}

it("renders ticket item with end date", () => {
  const ticket = {
    title: "General Admission",
    register: null,
    price: "$25.00",
    min: 1,
    max: 10,
    name: "ticket1",
    saleEndDate: new Date(1478708162000),
  };
  const tree = renderer.create(<TicketItem {...ticket} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders ticket item without end date", () => {
  const ticket = {
    title: "General Admission",
    register: null,
    price: "$25.00",
    min: 1,
    max: 12,
    name: "ticket1",
  };
  const tree = renderer.create(<TicketItem {...ticket} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders ticket item with badge", () => {
  const ticket = {
    register: null,
    title: "Early Bird",
    price: "$49.00",
    min: 5,
    max: 10,
    name: "ticket3",
    badge: { label: "Sold Out", type: "warning" },
  };
  const tree = renderer.create(<TicketItem {...ticket} />).toJSON();
  expect(tree).toMatchSnapshot();
});
