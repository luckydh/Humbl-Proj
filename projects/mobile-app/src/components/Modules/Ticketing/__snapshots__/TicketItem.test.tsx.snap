// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ticket item with badge 1`] = `
<div
  className="border-b border-gray-200 mb-6 pb-6 text-white"
>
  <div>
    <div
      className="flex w-full justify-between"
    >
      <div
        className="mr-4"
        style={
          Object {
            "maxWidth": "13rem",
          }
        }
      >
        <h3
          className="font-bold text-xl"
        >
           
          Early Bird
        </h3>
        <h4
          className="text-lg my-1"
        >
          $49.00
        </h4>
        <div />
      </div>
      <div>
        <span
          className="inline-flex items-center px-4 py-1 rounded-full text-base font-medium bg-red-light text-red"
        >
          Sold Out
        </span>
      </div>
    </div>
    <div />
  </div>
</div>
`;

exports[`renders ticket item with end date 1`] = `
<div
  className="border-b border-gray-200 mb-6 pb-6 text-white"
>
  <div>
    <div
      className="flex w-full justify-between"
    >
      <div
        className="mr-4"
        style={
          Object {
            "maxWidth": "13rem",
          }
        }
      >
        <h3
          className="font-bold text-xl"
        >
           
          General Admission
        </h3>
        <h4
          className="text-lg my-1"
        >
          $25.00
        </h4>
        <p
          className="text-base"
        >
          ticketing.component.ticket_item.sales_end_on
           
          Wednesday, Nov 9, 2016
        </p>
        <div />
      </div>
      <div>
        <select
          className="py-2 px-4 w-full rounded-lg font-semibold bg-blue border-white border-1 outline-none text-white placeholder-white-faded text-xl block focus:ring-indigo-500 focus:border-white focus:border-transparent focus:outline-none focus:ring-0 focus:placeholder-transparent"
          name="ticket1"
          style={
            Object {
              "fontSize": 20,
              "height": 47,
              "minWidth": 84,
            }
          }
        >
          <option>
            0
          </option>
          <option>
            1
          </option>
          <option>
            2
          </option>
          <option>
            3
          </option>
          <option>
            4
          </option>
          <option>
            5
          </option>
          <option>
            6
          </option>
          <option>
            7
          </option>
          <option>
            8
          </option>
          <option>
            9
          </option>
          <option>
            10
          </option>
        </select>
      </div>
    </div>
    <div />
  </div>
</div>
`;

exports[`renders ticket item without end date 1`] = `
<div
  className="border-b border-gray-200 mb-6 pb-6 text-white"
>
  <div>
    <div
      className="flex w-full justify-between"
    >
      <div
        className="mr-4"
        style={
          Object {
            "maxWidth": "13rem",
          }
        }
      >
        <h3
          className="font-bold text-xl"
        >
           
          General Admission
        </h3>
        <h4
          className="text-lg my-1"
        >
          $25.00
        </h4>
        <div />
      </div>
      <div>
        <select
          className="py-2 px-4 w-full rounded-lg font-semibold bg-blue border-white border-1 outline-none text-white placeholder-white-faded text-xl block focus:ring-indigo-500 focus:border-white focus:border-transparent focus:outline-none focus:ring-0 focus:placeholder-transparent"
          name="ticket1"
          style={
            Object {
              "fontSize": 20,
              "height": 47,
              "minWidth": 84,
            }
          }
        >
          <option>
            0
          </option>
          <option>
            1
          </option>
          <option>
            2
          </option>
          <option>
            3
          </option>
          <option>
            4
          </option>
          <option>
            5
          </option>
          <option>
            6
          </option>
          <option>
            7
          </option>
          <option>
            8
          </option>
          <option>
            9
          </option>
          <option>
            10
          </option>
          <option>
            11
          </option>
          <option>
            12
          </option>
        </select>
      </div>
    </div>
    <div />
  </div>
</div>
`;
